citrination_client/data/client.py

```diff
@@ -39,23 +39,29 @@ def __init__(self, api_key, host="https://citrination.com", suppress_warnings=Fa
         ]
         super(DataClient, self).__init__(api_key, host, members, suppress_warnings, proxies)
 
-    def upload(self, dataset_id, source_path, dest_path=None):
+    def upload(self, dataset_id, source_path, dest_path=None, is_async=True, timeout=30):
```

Comment:
could we call this flag `block_until_complete` or similar? It just isn't intuitive quite intuitive to me. `is_async=True` implies that I'll get back a UUID from this (and changing the return type based on an input parameter value seems very strange).

Reply:
Great point -- I think a lot of the confusion stemmed from the weird naming I picked. The boolean flags actually work the reverse of how I think you were reading it. Using your name makes it very obvious when the calls are blocking or not.